DuckDuckGo/DaxDialogs.swift

```diff
@@ -166,14 +177,46 @@ class DaxDialogs {
     func resumeRegularFlow() {
         nextHomeScreenMessageOverride = nil
     }
+    
+    private var fireButtonPulseTimer: Timer?
+    private static let timeToFireButtonExpire: TimeInterval = 1 * 60 * 60
+    
+    func fireButtonPulseStarted() {
+        if settings.fireButtonPulseDateShown == nil {
+            settings.fireButtonPulseDateShown = Date()
+            Pixel.fire(pixel: .fireEducationPulseShown)
+        }
+        if fireButtonPulseTimer == nil, let date = settings.fireButtonPulseDateShown {
+            let timeSinceShown = Date().timeIntervalSince(date)
+            let timerTime = DaxDialogs.timeToFireButtonExpire - timeSinceShown
+            fireButtonPulseTimer = Timer(timeInterval: timerTime, repeats: false) { _ in
+                self.settings.fireButtonEducationShownOrExpired = true
+                ViewHighlighter.hideAll()
+                if !self.settings.fireButtonEducationShownOrExpired {
+                    Pixel.fire(pixel: .fireEducationPulseCancelledBecauseTimeout)
```

Comment:
I think this didn't get called because you've already set the setting in the lines above.

Reply:
Ah thanks, good spot! I fixed something else and in the process broke this. Will fix